game/game.go

```diff
@@ -454,6 +465,32 @@ func (g *Game) GetPermanents(ids []PermanentId) []*Permanent {
 	return answer
 }
 
+// All objects must be put on the Stack via this method to get a unique ID.
+func (g *Game) AddToStack(stackObject *StackObject) {
+	stackObject.Id = g.NextStackObjectId
+	g.StackObjects[stackObject.Id] = stackObject
+	g.NextStackObjectId++
+	g.Stack = append(g.Stack, stackObject.Id)
+}
+
+func (g *Game) StackObject(id StackObjectId) *StackObject {
+	if id == StackObjectId(0) {
+		panic("0 is not a valid StackObjectId")
+	}
+	if g.StackObjects[id] == nil {
```

Comment:
more idiomatic and a teeny bit faster because it only needs one hash lookup would be, `obj, ok = g.StackObjects[id]`, and then panicking if !ok, and then returning `obj`